lib/semian/circuit_breaker.rb

```diff
@@ -1,13 +1,24 @@
 module Semian
-  class CircuitBreaker
-    attr_reader :state
-
-    def initialize(exceptions:, success_threshold:, error_threshold:, error_timeout:)
+  class CircuitBreaker #:nodoc:
+    def initialize(name, exceptions:, success_threshold:, error_threshold:, error_timeout:, permissions:)
+      @name = name.to_s
       @success_count_threshold = success_threshold
       @error_count_threshold = error_threshold
       @error_timeout = error_timeout
       @exceptions = exceptions
-      reset
+
+      @errors = ::Semian::SlidingWindow.new(@error_count_threshold,
+                                            name: "#{name}_sliding_window",
+                                            permissions: permissions)
+      @successes = ::Semian::AtomicInteger.new(name: "#{name}_atomic_integer",
+                                               permissions: permissions)
+      @state = ::Semian::AtomicEnum.new([:closed, :half_open, :open],
+                                        name: "#{name}_atomic_enum",
+                                        permissions: permissions)
+      # We do not need to #reset here since initializing is handled like this:
+      # (0) if data is not shared, then it's zeroed already
+      # (1) if no one is attached to the memory, zero it
+      # (2) otherwise, keep the data
```

Comment:
This comment is gonna get outdated... also I don't understand it... "if no one is attached to the memory, zero it"? what does attached mean? and "who" would be attached to things? this is very confusing, imho